frontend/javascripts/types/schemas/dataset_view_configuration_defaults.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import {
+  getDefaultLayerViewConfiguration,
+  defaultDatasetViewConfiguration,
+} from "types/schemas/dataset_view_configuration.schema";
+import { type APIDataset, type APIDataLayer } from "types/api_flow_types";
+import { getDefaultIntensityRangeOfLayer } from "oxalis/model/accessors/dataset_accessor";
+import { validateObjectWithType } from "types/validation";
+
+const eliminateErrors = (instance: Object, errors: Array<*>, defaults: Object) => {
+  errors.forEach(error => {
+    if (error.name === "required") {
+      instance[error.argument] = defaults[error.argument];
+    } else if (error.name === "additionalProperties") {
+      delete instance[error.argument];
+    } else {
+      const invalidFieldName = error.property.split(".")[1];
```

Comment:
This line assumes that the incorrect field name is in the first level of the object. If this was used for a nested structure like this:
```
{
  someProperty: {
    someInvalidProperty: 1
  }
}
```

the enforcement would remove `someProperty`, too, if I read this right. We probably don't have any nesting right now, but to avoid future problems, I suggest to do one of the following:
- assert that `error.property.split(".").length === 2` so that this will throw an error if we should have deeper nesting somewhere
- handle the depth correctly. not sure whether you can test this easily? Also note: It would be important that the library reports `someProperty.invalid1.invalid2.invalid3` with `error.property === "someProperty.invalid1"`. If it reports `invalid1.invalid2.invalid3`, then removing the last entry would not be enough.

Reply:
I think, this is fine because if there is a nested property, its type will be `object`, which is disallowed for every property except `layers`, which are corrected elsewhere. Therefore, the incorrect property will always be in the top-level. If we have nested properties in the future, I suggest handling them explicitly like the layers or adapt the code then.